src/Components/Main/main.tsx

```diff
@@ -1,12 +1,29 @@
-import React, { FC, ReactElement } from 'react';
+import React, { ReactElement, useEffect } from 'react';
 import { MovieCard } from '../movieCard/movieCard';
-import { MainProps } from './main.types';
-import { DataMovie } from '../../data/data.types';
+import { DataMovie } from '../../store/movies/movies.types';
 import './main.style.scss';
+import { getMovieById, loadMovies } from '../../store/movies/movies.actions';
+import { useDispatch, useSelector } from 'react-redux';
+
+export const Main = (): ReactElement => {
+    const sortBy = useSelector((store) => store.sortBy);
+    const searchBy = useSelector((store) => store.searchBy);
+    const searchInputValue = useSelector((store) => store.searchInput);
+    const movies = useSelector((store) => store.movies);
```

Comment:
Memoization of useSelector is nos not work with a new link to selectorFn. If you want to memorize  sortBy you need 
to move (store) => store.sortBy to separated function out the component like
const selectSortBy = (store) => store.sortBy;

Is applicable to all your selector function in this review.